vispy/visuals/visual.py

```diff
@@ -37,6 +38,24 @@ def __init__(self):
                                    update=Event,
                                    bounds_change=Event,
                                    )
+        self._gl_state = {'preset': None}
+
+    def set_gl_state(self, preset=None, **kwargs):
+        """Completely define the set of GL state parameters to use when drawing
+        this visual.
+        """
+        self._gl_state = kwargs
+        self._gl_state['preset'] = preset
+    
+    def update_gl_state(self, *args, **kwargs):
+        """Modify the set of GL state parameters to use when drawing
+        this visual.
+        """
+        if len(args) == 1:
+            self._gl_state['preset'] = args[0]
+        elif len(args) != 0:
+            raise TypeError("Only one positional argument allowed.")
+        self._gl_state.update(kwargs)
```

Comment:
No; that would cause `preset` to be changed even if the user does not request it.